bindings/experimental/distrdf/python/DistRDF/Profiling/__init__.py

```diff
@@ -0,0 +1,76 @@
+#  @author Giulio Crognaletti
+#  @author Vincenzo Eduardo Padulano
+#  @author Enric Tejedor
+#  @date 2022-08
+
+################################################################################
+# Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.                      #
+# All rights reserved.                                                         #
+#                                                                              #
+# For the licensing terms see $ROOTSYS/LICENSE.                                #
+# For the list of contributors see $ROOTSYS/README/CREDITS.                    #
+################################################################################
+
+# Inform cling to activate dumping of profiling information for jitted code.
+# It is crucial that this instruction is executed before importing ROOT in the distributed workers.
+from __future__ import annotations
+
+# TODO: Each time the module DistRDF is imported, the CLING_PROFILE=1 variable is set
+# modyfing the environment in both the client and the workers.
+# This affects only the workers when the profling option is enabled, since in all other
+# cases Cling is already initialized. However the environment is modified every time, and
+# this can potentially give rise to unexpected side effects.
+# The issue can be solved by adding the option to programmatically enable/disable 
+# Cling's profiling feature after its initialization, moving this step inside the 
+# data collection context manager for all Visualizations
+from os import environ
+environ["CLING_PROFILE"]="1"
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ..DataFrame import RDataFrame
+
+from .Flamegraph import FlameGraph
+
+SUPPORTED_VISUALIZATIONS = {
+    "flamegraph": FlameGraph
+}
+
+class ClingProfile():
+    """
+    Context manager to enable profiling of DistRDF applications.
+
+    All DistRDF code executed within the ClingProfile context will collect performance metric data in the workers and
+    merge them together in a single visualization (e.g. a flamegraph).
+    """
+
+    def __init__(self, rdf:RDataFrame, visualization:str="flamegraph", **options) -> None:
```

Comment:
One comment that just came to mind. Here the user can specify the visualization type, but how can they specify the profiler type?